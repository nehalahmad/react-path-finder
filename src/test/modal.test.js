import React from "react";
import Modal from "./../components/modal";
import { shallow } from "enzyme";

describe("<Modal />", () => {
  let setErrorMessageHandler;
  let errorMessage = "";
  beforeEach(() => {
    setErrorMessageHandler = jest.fn();
    errorMessage = "Oops! something went wrong!";
  });

  it("Render modal with a message", () => {
    const modalObj = shallow(
      <Modal
        errorMessage="Oops! something went wrong!"
        setErrorMessageHandler={setErrorMessageHandler}
      />
    );
    expect(modalObj).toMatchSnapshot();
  });

  it("Render modal without a message", () => {
    errorMessage = "";
    const modalObj = shallow(
      <Modal setErrorMessageHandler={setErrorMessageHandler} />
    );
    expect(modalObj).toMatchSnapshot();
  });

  it("it'll close the modal message", () => {
    const modalObj = shallow(
      <Modal
        errorMessage={errorMessage}
        setErrorMessageHandler={setErrorMessageHandler}
      />
    );
    expect(modalObj).toMatchSnapshot();

    let crossBtn = modalObj.find("button.close");
    expect(crossBtn.length).toEqual(1);

    crossBtn.simulate("click");
    expect(setErrorMessageHandler.mock.calls.length).toEqual(1);
  });
});
