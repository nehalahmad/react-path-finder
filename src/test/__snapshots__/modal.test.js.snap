// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Render modal with a message 1`] = `ShallowWrapper {}`;

exports[`<Modal /> Render modal without a message 1`] = `ShallowWrapper {}`;

exports[`<Modal /> Render toast with a message 1`] = `ShallowWrapper {}`;

exports[`<Modal /> Render toast without a message 1`] = `ShallowWrapper {}`;

exports[`<Modal /> it'll close the modal message 1`] = `ShallowWrapper {}`;

exports[`<Modal /> it'll close the toast message 1`] = `ShallowWrapper {}`;
